### PR TITLE
Added preconfigopts, configopts and a few other things in tensorflow …

### PR DESCRIPTION
**Pull request by damianam**
…easyblock

### CODE REVIEW

File: easybuild/easyblocks/t/tensorflow.py [hunk 1]
```diff
@@ -213,13 +219,21 @@ def build_step(self):
...
+            cmd.insert(0, 'export TF_MKL_DOWNLOAD=0 &&' )
+            cmd.insert(0, 'export TF_MKL_ROOT=%s &&' % mkl_root )
+        elif self.cfg['with_mkl_dnn']:
+            # this makes TensorFlow use mkl-dnn (cfr. https://github.com/01org/mkl-dnn)
+            cmd.extend(['--config=mkl'])
+            cmd.insert(0, 'export TF_MKL_DOWNLOAD=1 &&' )
 
         cmd.append('//tensorflow/tools/pip_package:build_pip_package')
 
+        cmd.insert(0, self.cfg['prebuildopts'])
```
Comment (boegel) [inline easybuild/easyblocks/t/tensorflow.py]: @damianam Please do this in the `cmd = ` line?

File: easybuild/easyblocks/t/tensorflow.py [hunk 2]
```diff
@@ -126,7 +127,12 @@ def configure_step(self):
         regex_subs = [(r"(run_shell\(\['bazel')", r"\1, '--output_base=%s'" % tmpdir)]
         apply_regex_substitutions('configure.py', regex_subs)
 
-        run_cmd('./configure', log_all=True, simple=True)
+        cmd = "%(preconfigopts)s ./configure %(configopts)s" % {
+            'preconfigopts': self.cfg['preconfigopts'],
+            'configopts': self.cfg['configopts'],
+        }
```
Comment (boegel) [inline easybuild/easyblocks/t/tensorflow.py]: this looks more complex than needed, how about:

```python
cmd = self.cfg['preconfigopts'] +  './configure ' + self.cfg['configopts']
run_cmd(cmd, log_all=True, simple=True)

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/t/tensorflow.py [hunk 3]
```diff
@@ -100,7 +101,7 @@ def configure_step(self):
             'TF_NEED_GCP': '0',  # Google Cloud Platform
             'TF_NEED_GDR': '0',
             'TF_NEED_HDFS': '0',  # Hadoop File System
-            'TF_NEED_JEMALLOC': ('0', '1')[bool(jemalloc_root)],
+            'TF_NEED_JEMALLOC': ('0', '1')[self.cfg['with_jemalloc']],
```
Comment (boegel) [inline easybuild/easyblocks/t/tensorflow.py:104]: And then what if `jemalloc` is not included as a dep? It'll auto-download?

Comment (damianam) [reply easybuild/easyblocks/t/tensorflow.py:104]: Yes. And actually, I think bazel just ignores the external installation. At least it does with MKL, even if you use `TF_MKL_ROOT`......

File: easybuild/easyblocks/t/tensorflow.py (hunk 2 above)
Comment (damianam) [reply easybuild/easyblocks/t/tensorflow.py]: Looks reasonable. I just followed the same approach as in other easyblocks.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Thanks for your help on taming this @damianam!

**Pull request merged**